homeassistant/components/pvpc_hourly_pricing/config_flow.py

```diff
@@ -96,14 +90,24 @@ async def async_step_user(
         )
         return self.async_show_form(step_id="user", data_schema=data_schema)
 
-    async def async_step_api_token(self, user_input: dict[str, Any]) -> FlowResult:
+    async def async_step_api_token(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
         """Handle optional step to define API token for extra sensors."""
-        self._api_token = user_input[CONF_API_TOKEN]
-        return await self._async_verify(
-            "api_token",
+        if user_input is not None and user_input.get(CONF_API_TOKEN):
```

Comment:
Remove the second condition as we know that the api token is required in the form if there's input.

Reply:
Done